test/models/requests/CreateUpdateExpensesRequestSpec.scala

```diff
@@ -0,0 +1,210 @@
+           |	"nino": "${user.nino}",
+           |	"mtditid": "${user.mtditid}",
+           |	"employmentExpensesData": {}
+           |}""".stripMargin)
+      Json.toJson(actualResult) shouldBe expectedJson
+    }
+
+    "creates an audit event" when {
+      "all prior expenses are provided" in {
+        val actualResult = defaultModel.toAmendAuditModel(taxYear, defaultPriorCustomerEmploymentExpenses)
+        val expectedJson = Json.parse(
+          s"""{
+             |	"taxYear": $taxYear,
+             |	"userType": "${user.affinityGroup.toLowerCase}",
+             |	"nino": "${user.nino}",
+             |	"mtditid": "${user.mtditid}",
+             |  "priorEmploymentExpensesData": {
+             |		"jobExpenses": 10,
+             |		"flatRateJobExpenses": 5,
+             |		"professionalSubscriptions": 14,
+             |		"otherAndCapitalAllowances": 21
+             |  },
+             |	"employmentExpensesData": {
+             |		"jobExpenses": 100,
+             |		"flatRateJobExpenses": 50,
+             |		"professionalSubscriptions": 140,
+             |		"otherAndCapitalAllowances": 210
+             |	}
+             |}""".stripMargin)
+        Json.toJson(actualResult) shouldBe expectedJson
+      }
+      "some prior expenses are provided " in {
+
+        val priorCustomerEmploymentExpenses = employmentExpenses.copy(
+          expenses = Some(defaultExpenses.copy(
+            businessTravelCosts = Some(15),
+            jobExpenses = Some(10),
+            flatRateJobExpenses = Some(5),
+            professionalSubscriptions = None,
+            hotelAndMealExpenses = Some(12),
+            otherAndCapitalAllowances = None,
+            vehicleExpenses = Some(25),
+            mileageAllowanceRelief = Some(30)
+          )))
+
+
+        val actualResult = defaultModel.toAmendAuditModel(taxYear, priorCustomerEmploymentExpenses)
+
+
+        val expectedJson = Json.parse(
+          s"""{
+             |	"taxYear": $taxYear,
+             |	"userType": "${user.affinityGroup.toLowerCase}",
+             |	"nino": "${user.nino}",
+             |	"mtditid": "${user.mtditid}",
+             |  "priorEmploymentExpensesData": {
+             |		"jobExpenses": 10,
+             |		"flatRateJobExpenses": 5
+             |  },
+             |	"employmentExpensesData": {
+             |		"jobExpenses": 100,
+             |		"flatRateJobExpenses": 50,
+             |		"professionalSubscriptions": 140,
+             |		"otherAndCapitalAllowances": 210
+             |	}
+             |}
+             |""".stripMargin)
+        Json.toJson(actualResult) shouldBe expectedJson
+      }
+      "some new expenses are provided" in {
+        val model = defaultModel.copy(
+          expenses = defaultExpenses.copy(
+            professionalSubscriptions = None,
+            otherAndCapitalAllowances = None
+          )
+        )
+
+        val actualResult = model.toAmendAuditModel(taxYear, defaultPriorCustomerEmploymentExpenses)
```

Comment:
Looks good

For completeness could you add another test (like for create) where the amend has no expenses too which will prove hey are all removed if they don't exist for this event too